### PR TITLE
Get the default glyph index once in DefaultCharacter {get}

### PR DESCRIPTION
**Pull request by nkast**

### CODE REVIEW

File: MonoGame.Framework/Graphics/SpriteFont.cs [hunk 1]
```diff
@@ -117,7 +119,19 @@ public int GetHashCode(char b)
...
-		public char? DefaultCharacter { get; set; }
+		public char? DefaultCharacter
+        {
+            get { return _defaultCharacter; }
+            set
+            {
+                _defaultCharacter = value;
+                // Get the default glyph index here once.
+                if (_defaultCharacter.HasValue)
+                    TryGetGlyphIndex(_defaultCharacter.Value, out _defaultGlyphIndex);
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/SpriteFont.cs]: What happens here if the user assigns a character not in the font?

Comment (nkast) [reply MonoGame.Framework/Graphics/SpriteFont.cs]: TryGetGlyphIndex() will set _defaultGlyphIndex = -1, that's all. 
No Exception is thrown until GetGlyphIndexOrDefault() tries to draw the default character.
I tried to preserve the existing behavior and it pass the existing tests.
Perhaps XNA throws a differed message in that case, I 'll have to check ...

Comment (nkast) [reply MonoGame.Framework/Graphics/SpriteFont.cs]: OK, XNA throws an ArgumentException immediately when you set DefaultCharacter.
We have to add a test for this change too, I will leave this for later.

**Comment by tomspilman** [comment]
This is great... thanks @nkast !

**Pull request merged**